nntrainer/layers/layer_node.h

```diff
@@ -370,40 +375,42 @@ class LayerNode final : public ml::train::Layer, public GraphNode {
    */
   void setOutputLayers(const std::vector<std::string> &layers) {
     output_layers = layers;
-    init_context =
-      InitLayerContext(init_context.getInputDimensions(),
-                       std::max((unsigned int)output_layers.size(), 1u));
   }
 
+  /**
+   * @brief check if input shape property is set
+   *
+   * @return bool true if input shape property has set
+   */
+  bool hasInputShapeProperty() const;
+
   /**
    * @brief Get the input dimension
    * @return TensorDim dimension of the input
    */
-  const std::vector<TensorDim> getInputDimensions() const {
-    return init_context.getInputDimensions();
-  }
+  const std::vector<TensorDim> getInputDimensions() const;
 
   /**
    * @brief Get the output dimension
    * @return TensorDim dimension of the output
    */
-  const std::vector<TensorDim> getOutputDimensions() const {
-    return init_context.getOutputDimensions();
-  }
-
+  const std::vector<TensorDim> getOutputDimensions() const;
   /**
    * @brief Get the Weight object
    *
    * @param idx Identifier of the weight
    * @return Weight& Reference to the weight
    */
   Weight getWeightWrapper(unsigned int idx) {
-    if (run_context.weightHasGradient(idx)) {
-      return Weight(run_context.getWeight(idx), run_context.getWeightGrad(idx),
-                    run_context.getWeightName(idx));
+    NNTR_THROW_IF(!run_context, std::runtime_error)
+      << __func__ << " layer node does not have the layer context yet!";
```

Comment:
how about "layer needs to be finalized first."?

Reply:
Done :)